Library/Homebrew/formula.rb

```diff
@@ -730,12 +735,32 @@ def bottle_prefix
     prefix+".bottle"
   end
 
-  # The directory where the formula's installation logs will be written.
+  # The directory where the formula's installation or test logs will be written.
   # @private
   def logs
     HOMEBREW_LOGS+name
   end
 
+  # The prefix, if any, to use in filenames for logging current activity
+  def active_log_prefix
+    if active_log_type
+      "#{active_log_type}."
+    else
+      ""
+    end
+  end
+
+  # Runs a block with the given log type in effect for its duration
+  def with_logging(log_type)
+    old_log_type = @active_log_type
+    @active_log_type = log_type
+    begin
+      yield
+    ensure
+      @active_log_type = old_log_type
+    end
+  end
```

Comment:
I think that's true, and it tested out when I tried it. Switched to that form. Look better?